File: common/src/main/java/com/github/xiaogegechen/common/adapter/MyFragmentPagerAdapter.java
package com.github.xiaogegechen.common.adapter;

import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.github.xiaogegechen.common.base.BaseFragment;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public abstract class MyFragmentPagerAdapter extends FragmentPagerAdapter {
    private List<BaseFragment> mFragmentList;
    private FragmentManager mFragmentManager;

    public MyFragmentPagerAdapter(FragmentManager fm, List<BaseFragment> fragmentList) {
        super(fm);
        mFragmentManager = fm;
        mFragmentList = fragmentList;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList == null ? null : mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList == null ? 0 : mFragmentList.size();
    }

    @NotNull
    @Override
    public Object instantiateItem(@NotNull ViewGroup container, int position) {
        Fragment fragment = (Fragment)super.instantiateItem(container,position);
        mFragmentManager.beginTransaction().show(fragment).commit();
        return fragment;
    }

    @Override
    public void destroyItem(@NotNull ViewGroup container, int position, @NotNull Object object) {
        Fragment fragment = mFragmentList.get(position);
        mFragmentManager.beginTransaction().hide(fragment).commit();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        if(getTitle () != null){
            return getTitle ().get (position);
        }
        return null;
    }

    //设置title,如果不是和TabLayout绑定,返回null
    public abstract List<String> getTitle();
}

File: common/src/main/java/com/github/xiaogegechen/common/adapter/MyFragmentPagerAdapter.kt
package com.github.xiaogegechen.common.adapter

import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.github.xiaogegechen.common.base.BaseFragment

abstract class MyFragmentPagerAdapter(private val mFragmentManager: FragmentManager,
                                      private val mFragmentList: MutableList<BaseFragment>?): FragmentPagerAdapter(mFragmentManager) {

    override fun getItem(position: Int): Fragment? {
        return mFragmentList?.get(position)
    }

    override fun getCount(): Int {
        return mFragmentList?.size ?: 0
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val fragment =  super.instantiateItem(container, position) as Fragment
        mFragmentManager.beginTransaction().show(fragment).commit()
        return fragment
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        val fragment = mFragmentList?.get(position)!!
        mFragmentManager.beginTransaction().hide(fragment).commit()
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return getTitle()?.get(position)
    }

    // 设置title,如果不是和TabLayout绑定,返回null
    abstract fun getTitle(): List<String>?
}